sonar-application/src/main/java/org/sonar/application/App.java

```diff
@@ -36,63 +39,68 @@
 
 import static org.sonar.process.Lifecycle.State;
 import static org.sonar.process.ProcessId.APP;
-import static org.sonar.process.ProcessProperties.HTTPS_PROXY_HOST;
-import static org.sonar.process.ProcessProperties.HTTPS_PROXY_PORT;
-import static org.sonar.process.ProcessProperties.HTTP_PROXY_HOST;
-import static org.sonar.process.ProcessProperties.HTTP_PROXY_PORT;
 
 /**
  * Entry-point of process that starts and monitors ElasticSearch, the Web Server and the Compute Engine.
  */
 public class App implements Stoppable {
 
-  /**
-   * Properties about proxy that must be set as system properties
-   */
-  private static final String[] PROXY_PROPERTY_KEYS = new String[] {
-    HTTP_PROXY_HOST,
-    HTTP_PROXY_PORT,
-    "http.nonProxyHosts",
-    HTTPS_PROXY_HOST,
-    HTTPS_PROXY_PORT,
-    "http.auth.ntlm.domain",
-    "socksProxyHost",
-    "socksProxyPort"};
-
+  private final Properties commandLineArguments;
+  private final Function<Properties, Props> propsSupplier;
+  private final JavaCommandFactory javaCommandFactory;
   private final Monitor monitor;
+  private final Supplier<List<JavaCommand>> javaCommandSupplier;
+
+  private App(Properties commandLineArguments) {
+    this.commandLineArguments = commandLineArguments;
+    this.propsSupplier = properties -> new PropsBuilder(properties, new JdbcSettings()).build();
+    this.javaCommandFactory = new JavaCommandFactoryImpl();
+    Props props = propsSupplier.apply(commandLineArguments);
+
+    AppFileSystem appFileSystem = new AppFileSystem(props);
+    appFileSystem.verifyProps();
+    AppLogging logging = new AppLogging();
+    logging.configure(props);
 
-  public App(AppFileSystem appFileSystem, boolean watchForHardStop) {
-    this(Monitor.newMonitorBuilder()
+    // used by orchestrator
+    boolean watchForHardStop = props.valueAsBoolean(ProcessProperties.ENABLE_STOP_COMMAND, false);
+    this.monitor = Monitor.newMonitorBuilder()
       .setProcessNumber(APP.getIpcIndex())
       .setFileSystem(appFileSystem)
       .setWatchForHardStop(watchForHardStop)
       .setWaitForOperational()
       .addListener(new AppLifecycleListener())
-      .build());
+      .build();
+    this.javaCommandSupplier = new ReloadableCommandSupplier(props, appFileSystem::ensureUnchangedConfiguration);
   }
 
-  App(Monitor monitor) {
+  @VisibleForTesting
+  App(Properties commandLineArguments, Function<Properties, Props> propsSupplier, Monitor monitor, CheckFSConfigOnReload checkFsConfigOnReload, JavaCommandFactory javaCommandFactory) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Split this 184 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)